packages/expo-cli/src/commands/build/index.ts

```diff
@@ -49,6 +49,7 @@ export default function(program: Command) {
       '--public-url <url>',
       'The URL of an externally hosted manifest (for self-hosted apps).'
     )
+    .option('--skip-credentials-check', 'Skip checking credentials.')
```

Comment:
I'd also hide this option from the help output. However, it looks like it's not possible to do it with commander yet - https://github.com/tj/commander.js/issues/1106 :(